pennylane_qiskit/converter.py

```diff
@@ -391,22 +396,91 @@ def _function(*args, params: dict = None, wires: list = None, **kwargs):
                             meas_terminal = False
                             break
 
+                # Allows for adding terminal measurements
+                if meas_terminal:
+                    terminal_meas.extend(operation_wires)
+
                 # Allows for queing the mid-circuit measurements
-                if not meas_terminal:
-                    mid_circ_meas.append(qml.measure(wires=operation_wires))
                 else:
-                    terminal_meas.extend(operation_wires)
+                    operation_func = qml.measure
+                    mid_circ_meas.append(qml.measure(wires=operation_wires))
+
+                    # Allows for tracking conditional operations
+                    for carg in cargs:
+                        mid_circ_regs[carg] = mid_circ_meas[-1]
+
+            # TODO: this may contain some logic for the bigger ControlFlowOps
+            elif isinstance(ops, ControlFlowOp):
+                operation_cond = True
 
             else:
+
                 try:
-                    operation_matrix = op.to_matrix()
-                    pennylane_ops.QubitUnitary(operation_matrix, wires=operation_wires)
+                    operation_args = [ops.to_matrix()]
+                    operation_func = qml.QubitUnitary
+
                 except (AttributeError, QiskitError):
                     warnings.warn(
                         f"{__name__}: The {instruction_name} instruction is not supported by PennyLane,"
                         " and has not been added to the template.",
                         UserWarning,
                     )
+
+            # Check if it is a conditional operation
+            if operation_cond or (ops.condition and ops.condition[0] in mid_circ_regs):
+                # Iteratively recurse over to build different branches
+                with qml.QueuingManager.stop_recording():
+                    branch_funcs = [
+                        partial(load(branch_inst, measurements=None), params=params, wires=wires)
+                        for branch_inst in operation_params
+                        if isinstance(branch_inst, QuantumCircuit)
+                    ]
```

Comment:
In reality, it's just another parameter. I am calling it `branch_inst` because it is an `Instruction` on what to do in the conditional branch. About the `branch_funcs` - I wanted to indicate that each function corresponds to a specific branch which `qml.cond` would use. I can call it `branch_qfuncs`.